code/modules/atmospherics/machinery/components/unary_devices/airlock_pump.dm

```diff
@@ -0,0 +1,455 @@
+	///The start time of the current cycle to calculate cycle duration
+	var/cycle_start_time
+	///Max duration of cycle, after which the pump will unlock the airlocks with a warning
+	var/cycle_timeout = 10 SECONDS
+	///List of the turfs adjacent to the pump for faster cycling and avoiding wind
+	var/list/turf/adjacent_turfs = list()
+	///Max distance between the airlock and the pump. Used to set up cycling.
+	var/airlock_pump_distance_limit = 2
+	///Max distance between the central airlock and the side airlocks in a group
+	var/airlock_group_distance_limit = 2
+	///Type of airlocks required for automatic cycling setup. To avoid hacking bridge doors. Ignored for mapspawn pump.
+	var/valid_airlock_typepath = /obj/machinery/door/airlock/external
+	///Station-facing airlocks used in cycling
+	var/list/obj/machinery/door/airlock/internal_airlocks
+	///Space-facing airlocks used in cycling
+	var/list/obj/machinery/door/airlock/external_airlocks
+	///Whether both airlocks are specified and cycling is available
+	var/cycling_set_up = FALSE
+	///Whether the pump opens the airlocks up instead of simpy unbolting them on cycle
+	var/open_airlock_on_cycle = TRUE
+	///Airlocks currently animating
+	var/airlocks_animating = FALSE
+	///Whether the airlocks comment the cycling details to the chat
+	var/is_cycling_audible = TRUE
+
+	COOLDOWN_DECLARE(check_turfs_cooldown)
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/update_icon_nopipes()
+	if(!on || !is_operational || !powered())
+		icon_state = "vent_off"
+	else
+		icon_state = pump_direction ? "vent_out" : "vent_in"
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/update_overlays()
+	. = ..()
+	if(!showpipe)
+		return
+
+	var/mutable_appearance/distro_pipe_appearance = get_pipe_image(icon, "pipe_exposed", dir, COLOR_BLUE, piping_layer = 4)
+	if(nodes[1])
+		distro_pipe_appearance = get_pipe_image(icon, "pipe_intact", dir, COLOR_BLUE, piping_layer = 4)
+	. += distro_pipe_appearance
+
+	var/mutable_appearance/waste_pipe_appearance = get_pipe_image(icon, "pipe_exposed", dir, COLOR_RED, piping_layer = 2)
+	if(nodes[2])
+		waste_pipe_appearance = get_pipe_image(icon, "pipe_intact", dir, COLOR_RED, piping_layer = 2)
+	. += waste_pipe_appearance
+
+	var/mutable_appearance/distro_cap_appearance = get_pipe_image(icon, "vent_cap", dir, piping_layer = 4)
+	. += distro_cap_appearance
+
+	var/mutable_appearance/waste_cap_appearance = get_pipe_image(icon, "vent_cap", dir, piping_layer = 2)
+	. += waste_cap_appearance
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/atmos_init(list/node_connects)
+	for(var/obj/machinery/atmospherics/target in get_step(src, dir))
+		if(connection_check(target, 4) && !nodes[1])
+			nodes[1] = target // Distro
+		if(connection_check(target, 2) && !nodes[2])
+			nodes[2] = target // Waste
+	update_appearance()
+
+
+/obj/machinery/atmospherics/components/unary/airlock_pump/Initialize(mapload)
+	. = ..()
+	if(mapload)
+		can_unwrench = FALSE
```

Comment:
Why not? Just curious.

Reply:
1. Crafted pump can be connected to "external" airlock type only, while mapped ones connect to any airlocks. This is needed to avoid door hacking by placing crafted airlock pump, creating fake airlock.
2. To protect map-spawn pump somehow, I plan to mass-place it on some maps. This is needed to protect them from simple unwrench by a random guy.